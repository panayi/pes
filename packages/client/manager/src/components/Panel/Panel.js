import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { renameProp, compose } from 'recompose';
import withStyles from '@material-ui/core/styles/withStyles';
import Toolbar from '@material-ui/core/Toolbar';

const HEADER_HEIGHT = 64;

const styles = theme => ({
  root: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'flex-start',
    height: '100%',
    overflow: 'hidden',
  },
  header: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'flex-start',
    height: HEADER_HEIGHT,
    padding: [0, theme.spacing.unit * 2],
    borderBottom: `1px solid ${theme.palette.divider}`,
  },
  content: {
    position: 'relative',
    flex: 1,
    height: `calc(100% - ${HEADER_HEIGHT}px)`,
  },
});

// Renders a fixed header with scrollable content
const Panel = ({ header, children, classesFromProps, classes }) => (
  <div className={classNames(classes.root, classesFromProps.root)}>
    {header && (
      <Toolbar
        classes={{ root: classNames(classes.header, classesFromProps.header) }}
      >
        {header}
      </Toolbar>
    )}
    <div className={classNames(classes.content, classesFromProps.content)}>
      {children}
    </div>
  </div>
);

Panel.propTypes = {
  header: PropTypes.node,
  children: PropTypes.node,
  classesFromProps: PropTypes.shape({}),
  classes: PropTypes.shape({}).isRequired,
};

Panel.defaultProps = {
  header: null,
  children: null,
  classesFromProps: {},
};

export default compose(
  renameProp('classes', 'classesFromProps'),
  withStyles(styles),
)(Panel);
