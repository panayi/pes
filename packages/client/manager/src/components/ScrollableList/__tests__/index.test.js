import React from 'react';
import Panel from '../../Panel/Panel';
import ScrollableListHoc from '..';

describe('ScrollableList', () => {
  describe('Index', () => {
    it('should render properly', () => {
      const Component = shallow(<ScrollableListHoc />);

      expect(Component).toMatchSnapshot();
    });

    it('should pass extra props to the panel', () => {
      const someExtraProps = {
        header: <div className="header">header</div>,
      };
      const children = <div className="children">children</div>;
      const Component = mount(
        <ScrollableListHoc fullWidth {...someExtraProps}>
          {children}
        </ScrollableListHoc>,
      );
      const allExtraProps = { ...someExtraProps, children };

      expect(Component.find(Panel).props()).toEqual(allExtraProps);
    });
  });
});
