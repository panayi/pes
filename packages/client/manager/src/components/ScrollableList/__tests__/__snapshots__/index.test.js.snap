// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollableList Index should render properly 1`] = `
<WithStyles(Component)
  panelProps={Object {}}
/>
`;
